lib/src/main/java/com/basistheory/android/view/TextElement.kt

```diff
@@ -68,15 +71,29 @@ open class TextElement @JvmOverloads constructor(
         subscribeToInputEvents()
     }
 
-    // this MUST be internal to prevent host apps from accessing the raw input values
+    // the following getters MUST be internal to prevent host apps from accessing the raw input values
+
     internal fun getText(): String? =
+        editText.text?.toString()
+
+    internal fun getTransformedText(): String? =
         editText.text?.toString().let {
             transform?.apply(it) ?: it
         }
 
     fun setText(value: String?) =
         editText.setText(value)
 
+    private var valueRef: TextElement? = null
+    fun setValueRef(element: TextElement) {
+        if (element !== valueRef) {
```

Comment:
since the element can't be null, the reference can't be detached, but they shouldn't be doing that so maybe it's fine